### PR TITLE
Improve zebra-consensus logging and metrics

### PR DESCRIPTION
**Pull request by teor2345**
These logging and metrics changes help us diagnose sync and verify issues.

### CODE REVIEW

File: zebra-consensus/src/chain.rs [hunk 1]
```diff
@@ -170,6 +167,13 @@ where
                     .await?;
             }
 
+            tracing::trace!(?height, ?hash, "Verified block");
+            metrics::gauge!(
+                "chain.verified.block.height",
+                height.expect("Valid blocks have a block height").0 as _
+            );
+            metrics::counter!("chain.verified.block.count", 1);
+
```
Comment (teor2345) [inline zebra-consensus/src/chain.rs:176]: Actually, I'll want to delete this in #969, so I'd best do it now.
```suggestion
```

Comment (teor2345) [reply zebra-consensus/src/chain.rs:176]: Hmm actually no I want to keep this, so we have a single metric which covers checkpoints and blocks.

File: zebra-consensus/src/block.rs [hunk 2]
```diff
@@ -123,20 +123,35 @@ where
             // Skip contextual checks for the genesis block
             let previous_block_hash = block.header.previous_block_hash;
             if previous_block_hash != crate::parameters::GENESIS_PREVIOUS_BLOCK_HASH {
-                tracing::debug!(?height, "Awaiting previous block from state");
+                if height == block::Height(0) {
+                    Err("Invalid block: height is 0, but previous block hash is not null.")?;
```
Comment (yaahc) [inline zebra-consensus/src/block.rs:127]: This is a style that rustc / clippy use for errors but I think its a good one to adopt here as well, no capitalization or punctuation in the error messages:

```suggestion
                    Err("invalid block: height is 0, but previous block hash is not null")?;
```

Hmm, I wonder if clippy could lint for this...

File: zebra-consensus/src/block.rs [hunk 3]
```diff
@@ -123,20 +123,35 @@ where
...
                     previous_block_hash,
-                    block::Height(height.0 - 1),
+                    expected_height,
                 )
                 .await?;
 
                 let previous_height = previous_block.coinbase_height().unwrap();
-                if height.0 != previous_height.0 + 1 {
+                if previous_height != expected_height {
                     Err("Invalid block height: must be 1 more than the previous block height.")?;
```
Comment (yaahc) [inline zebra-consensus/src/block.rs:144]:
```suggestion
                    Err("invalid block height: must be 1 more than the previous block height")?;
```

File: zebra-consensus/src/chain.rs [hunk 4]
```diff
@@ -111,28 +111,25 @@ where
...
         // TODO(jlusby): Error = Report, handle errors from state_service.
-        let span = tracing::debug_span!(
-            "block_verify",
-            height = ?block.coinbase_height(),
-            hash = ?block.hash()
-        );
-        let block_height = block.coinbase_height();
+        let height = block.coinbase_height();
+        let hash = block.hash();
+        let span = tracing::debug_span!("block_verify", ?height, ?hash,);
```
Comment (yaahc) [inline zebra-consensus/src/chain.rs:116]: I think this needs to be entered unless I'm missing where it's used later.
```suggestion
        let span = tracing::debug_span!("block_verify", ?height, ?hash,);
        let _guard = span.enter();
```

**Review by yaahc — CHANGES_REQUESTED** [review]
looks good, I have two style comments and 1 actually important comment I think, if not for the 3rd comment I'd be approving this.

File: zebra-consensus/src/chain.rs (hunk 4 above)
Comment (teor2345) [reply zebra-consensus/src/chain.rs:116]: I think you're right, we must have missed that in a much earlier review.

Comment (teor2345) [reply zebra-consensus/src/chain.rs:116]: Actually, no, the span is used to instrument the future:
https://github.com/ZcashFoundation/zebra/pull/989/files#diff-9ce8cedb3963941e6cffe1a307d9dafcL183

Comment (yaahc) [reply zebra-consensus/src/chain.rs:116]: aah yea there it is, just outside of the diff, lol

**Review by yaahc — APPROVED** [review]

**Pull request merged**